src/stdatamodels/model_base.py

```diff
@@ -120,6 +121,15 @@ def __init__(self, init=None, schema=None, memmap=False,
             If `None`, value will be taken from the environmental STRICT_VALIDATION.
             Otherwise, the default value is `False`.
 
+        validate_on_assignment : bool or None
+            Defaults to 'None'.
+            If `None`, value will be taken from the environmental VALIDATE_ON_ASSIGNMENT,
+            defaulting to 'True' if  no environment variable is set.
+            If 'True', attribute assignments are validated at the time of assignment.
+            Validation errors generate warnings and values will be set to `None`.
```

Comment:
Doesn't the warning vs exception still depend on the value of `strict_validation`?  Or do we always get warnings when `validate_on_assignment` is set?